### ISSUE TITLE
Open Delete History Temp container tab from a link on a Multi-Account container tab

### ISSUE THREAD

**Issue by synthgab** [stoically/temporary-containers#84]
- Temporary Containers Version: 0.71
- Firefox Version: 59.0
- Operating System + Version: Windows 7
- Non-default Options/Preferences in Temporary Containers: one website pattern is set to demo the issue (www.nextinpact.com) , Isolation navigation is set to Never , all options in  Delete History are non-Default
- Other installed Add-ons + Version: Multi Account 6.0.0 , ublock 1.15.10 , skip redirects 2.2.1 , Smart Referer, HTTPS everywhere, Privacy BAdger, Cookie auto delete, KeeFox, Bloody Vikings

### Actual behavior
When clicking a link on a permanent tab from an assigned site (e.g. twitter), no matter what, a standard temporary container is always opened.

### Expected behavior
From my understanding, the option « Open New "Delete History Temporary Containers" for Website patterns configured in "Always per Website" instead of normal Temporary Containers » should open these links into Delete History containers...

### Steps to reproduce

1. Assign Twitter / Facebook / Reddit / etc to a container with Multi-Account container
2. Open a tab in the corresponding container, then login (Temp container are not yet involved)
3. In the recent publications, choose a link to an external site
4. Set this site in the Always per Website pattern list of Temporary Container configuration page
5. Click on the link : it'll open in a normal temporary container

### Notes
I personnaly don't consider it as critical or anything, I was just trying to figure out how to efficiently use patterns and delete history containers, and came across this behavior that I wasn't able to understand.

[debug.txt](https://github.com/stoically/temporary-containers/files/1788845/debug.txt)

**Comment by synthgab** [comment]
Sorry about that, after more investigations, it's related to another add-on (Neat URL , for further reference).

**Comment by synthgab** [comment]
Hello, please accept my deepest apologies for the back-and-forth comments, but now I have figured out what happens.

Actually, this behavior shows up when any other extension modifies the URL (Neat URL, Link Cleaner, Request Control, etc.), and with that in mind, we can see this behavior in the provided debug log.

Just an example to highlight the issue, with a URL cleaning extension:

1/ The new tab is successfully opened in a delete history temporary container (URL contains garbage)
2/ The interfering extensions intercepts the request, filters the URL, and re-issue the request (clean-up URL)
3/ At this point, the tab is once more re-opened in a temporary container, but this time, a normal one (why ? and is it possible to circumvent that ?)

It's a very specific use-case, but has to be known regarding privacy, because this means that at some point, a choie has to be made between keeping tracking URL parameters (utm_, ga_, ...) or open these links in delete history temporary containers.

Regards.

**Comment by stoically** [comment]
Hey, as I can see from your debug log (thanks for providing that!), you have the Multi-Account Containers Isolation activated

```
[shouldIsolateMac] mac isolating because request url is not assigned to the tabs permanent container  log.js:13:17
```

And there's currently no advanced setting to automatically create Deletes-History Temporary Containers when Isolation takes place. I already left me a TODO in the code :D

https://github.com/stoically/temporary-containers/blob/3e41bcab2b1177a037c8b46c914aad63454e7de3/src/background/request.js#L361

Wouldn't had thought that this exact combination would get actually used right off the bat, but I guess I was wrong.

Thanks for taking the time to open the Issue, I'll look into adding the preference ASAP.

**Comment by synthgab** [comment]
:D 

Thank you for the heads up and the efficient follow-up of your great extension :+1:

**Comment by stoically** [comment]
Just published version 0.72 with the advanced preferences to open deletes-history containers in case of Isolation. Let me know if something still doesn't work as expected.
